### DISCUSSION TITLE
reuse of knowledgeBase

### DISCUSSION THREAD

**Discussion by Leopord7** [hyperjumptech/grule-rule-engine#342] [Q&A] [answered]
Can we reuse the same knowledgeBase in different evaluation?
for example:
```

knowledgeBase := knowledgeLibrary.NewKnowledgeBaseInstance("TutorialRules", "0.0.1")

engine := engine.NewGruleEngine()

err = engine.Execute(dataCtx1, knowledgeBase)
err = engine.Execute(dataCtx2, knowledgeBase)
```

**Comment by Homeblest** [comment, marked as answer]
Yep!

**Reply by adarshinde** [reply]
Hey,
Can we reuse the same knowledgeBase for different evaluations, which are happening concurrently in multiple goroutines?

**Reply by newm4n** [reply]
You can see the sample and test ... examples/Concurrency_test.go
